src/main/scala/com/netflix/edda/aws/AwsCrawlers.scala

```diff
@@ -160,9 +156,13 @@ abstract class AwsIterator extends Iterator[Seq[Record]] {
 class AwsAddressCrawler(val name: String, val ctx: AwsCrawler.Context) extends Crawler {
   val request = new DescribeAddressesRequest
 
-  override def doCrawl()(implicit req: RequestId) =
-    backoffRequest { ctx.awsClient.ec2.describeAddresses(request).getAddresses.asScala.map(
+  private def doNonAssumeCrawl()(implicit req: RequestId) =
+    backoffRequest { ctx.awsClient.ec2().describeAddresses(request).getAddresses.asScala.map(
       item => Record(item.getPublicIp, ctx.beanMapper(item))) }.toSeq
+
+  override def doCrawl()(implicit req: RequestId) =
+    backoffRequest { ctx.awsClient.ec2(true).describeAddresses(request).getAddresses.asScala.map(
+      item => Record(item.getPublicIp, ctx.beanMapper(item))) }.toSeq ++ (if(assumeRoleEnabled) doNonAssumeCrawl() else Seq.empty[Record])
```

Comment:
this logic doesn't look correct, if assumeRoleEnabled == true then doNonAssumeCrawl()?

this structuring is pretty awkward, feels like the conditional should be handled up front or pushed down outside of this context.